react-app/src/app/screens/Game/index.js

```diff
@@ -1,58 +1,44 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import gameActions from '../../../redux/game/actions';
 
 import calculateWinner from './utils';
 import Board from './components/Board';
 import style from './styles.scss';
 
 class Game extends Component {
-  state = {
-    history: [{ squares: Array(9).fill(null) }],
-    stepNumber: 0,
-    xIsNext: true
-  };
-
   handleClick = i => {
-    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const history = this.props.history.slice(0, this.props.stepNumber + 1);
     const current = history[history.length - 1];
     const squares = current.squares.slice();
     if (calculateWinner(squares) || squares[i]) {
       return;
     }
-    squares[i] = this.state.xIsNext ? 'X' : 'O';
-    this.setState({
-      history: history.concat([{ squares }]),
-      stepNumber: history.length,
-      xIsNext: !this.state.xIsNext
-    });
+    squares[i] = this.props.xIsNext ? 'X' : 'O';
+    this.props.toggleMark(history.concat([{ squares }]), history.length, !this.props.xIsNext);
   };
 
   createHistoryButton(move) {
     const desc = move ? `Go to move # ${move}` : 'Go to game start';
     return (
       <li key={move.toString()}>
-        <button onClick={() => this.jumpTo(move)}>{desc}</button>
+        <button onClick={() => this.props.jumpTo(move, move % 2 === 0)}>{desc}</button>
       </li>
     );
   }
 
-  jumpTo = step => {
-    this.setState({
-      stepNumber: step,
-      xIsNext: step % 2 === 0
-    });
-  };
-
   render() {
-    const history = this.state.history;
-    const current = history[this.state.stepNumber];
+    const history = this.props.history;
+    const current = history[this.props.stepNumber];
     const winner = calculateWinner(current.squares);
```

Comment:
I would map the winner in redux. Think that you are calculating the winner in each render! If you moved this logic to the action, then you would only calculate the winner with each click.